lib/runner/test-runner/test-runner.js

```diff
@@ -42,50 +48,47 @@ module.exports = testRunner = {
 
     test.name = testPath.split('\\').pop().split('/').pop();
 
-        testRunner._testNumbers.hasOwnProperty(test.name) ?
-        test.number = testRunner._testNumbers[test.name] :
-        testRunner._testNumbers[test.name] = test.number = Object.keys(testRunner._testNumbers).length;
-
-        test.testPath = testPath;
-
-        testRunner._testsRunning++;
-
-        test.stderr.on('data', function(data) {
-          testRunner.emit('testRunner.childStderrReceived', data, test.number);
-        });
-
-        test.on('message', function(data) {
-          testRunner.emit('testRunner.testDataReceived', data, test.number);
-        });
-
-        test.on('close', function(exitCode) {
-          if (exitCode) {
-            if (testRunner._rerunCount === 0) {
-              process.exitCode = 1;
-            }
-            testRunner._testsToRerun.push(test.testPath);
-          }
-          testRunner._testsRunning--;
-          testRunner._testsRemaining--;
-          testRunner.emit('testRunner.testFinished', test.number, testRunner._rerunCount);
-          if (testRunner._testsRemaining === 0) {
-            if (testRunner._testsToRerun.length === 0 || testRunner._rerunCount === 0) {
-              testRunner.emit('testRunner.done');
-            } else {
-              testRunner.emit('testRunner.rerunTestsReadyToConfigure');
-            }
-          }
-        });
-
-        testRunner.emit('testRunner.testStarted', test.number, test.name);
-  },
-  _configureRerun() {
-    testRunner._rerunCount--;
-    testRunner._testFiles = testRunner._testsToRerun;
-    testRunner._testsToRerun = [];
-    testRunner._testsRemaining = testRunner._testFiles.length;
-    testRunner.emit('testRunner.configured');
+    testRunner._testNumbers.hasOwnProperty(test.name) ?
+      test.number = testRunner._testNumbers[test.name] :
+      testRunner._testNumbers[test.name] = test.number = Object.keys(testRunner._testNumbers).length;
+
+    test.testPath = testPath;
+
+    testRunner._testsRunning++;
+
+    test.stderr.on('data', function(data) {
+      testRunner.emit('testRunner.childStderrReceived', data, test.number);
+    });
+
+    test.on('message', function(data) {
+      testRunner.emit('testRunner.testDataReceived', data, test.number);
+    });
+
+    test.on('close', function(exitCode) {
+      if (exitCode) {
+        if (testRunner._rerunCount === 0) {
```

Comment:
What happens to this variable here? I don't think the exit code will be correctly set since the `_rerunCount` was is not longer decremented. We still need a way to se the exit code to 1 if tests have been rerun and they did not pass. Just from scanning this I think the when the `exitCode` comes in as non-zero and `testRunner._testsRerunMemo[test.testPath] === 0` then we set the `process.exitCode` to 1. Does that make sense?

Reply:
Or whatever way is best that you can think of for when all reruns for a test have been done and the last rerun failed we set the `process.exitCode` to 1.

Reply:
I can set it when tests print a failure state, as it will only do that if there are no more reruns

Reply:
Ok, just as long as it always runs regardless of which reporters are enabled. After looking at the code again I think here `https://github.com/GannettDigital/simulato/pull/216/files#diff-4c167b9bc744856bc98c96c70f1a0d46R80` where it emits finished. Or where finished is handled.

Reply:
I put it into finalizeTestReport, as that is where we check if a test is a rerun or not